src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,121 @@
+package duke;
+
+/**
+ * The Parser class handles the parsing of user input commands and performs
+ * corresponding actions.
+ */
+public class Parser {
+    public static final String COMMAND_TODO = "todo";
+    public static final String COMMAND_DEADLINE = "deadline";
+    public static final String COMMAND_EVENT = "event";
+    public static final String COMMAND_DELETE = "delete";
+    public static final String COMMAND_MARK = "mark";
+    public static final String COMMAND_UNMARK = "unmark";
+
+    /**
+     * Parses the user input and adds the corresponding task to the task list.
+     *
+     * @param userInput The user input command.
+     * @param taskList  The task list to which the task will be added.
+     * @throws IllegalArgumentException If the input command is invalid.
+     */
+    public static String parseAndAddTask(String userInput, TaskList taskList) {
+        String[] words = userInput.split(" ", 2);
+
+        switch (words[0].toLowerCase()) {
```

Comment:
Consider removing the indentation for the case clauses to adhere to coding standard.